pkg/kubectl/cmd/patch.go

```diff
@@ -200,14 +200,17 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 			}
 			count++
 
-			if err := info.Refresh(patchedObj, true); err != nil {
+			//info.Object is updated by the info.Refresh call.
+			//we should compute old data prior to refresh
+			oldData, err := json.Marshal(info.Object)
```

Comment:
do you think code [here]( https://github.com/kubernetes/kubernetes/blob/90c12fff84f88cf560fd4e34add14bc79090b20c/pkg/kubectl/cmd/edit.go#L448 ) could works ?  if not, I think maybe we should also change here.